Project.toml

```diff
@@ -15,9 +15,7 @@ Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [weakdeps]
 AxisKeys = "94b1ba4f-4ee9-5380-92f1-94cde586c3c5"
-Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 IntervalSets = "8197267c-284f-5f27-9208-e0e47529a953"
-LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
```

Comment:
AFAICT to break the cycle due to the loading behaviour on Julia 1.10 it would be sufficient to make one of Dates and LinearAlgebra a regular dependency. This is also how it was in the previous release it seems (only LinearAlgebra was added as a weak dependency). I would guess that LinearAlgebra triggers more and slower to load extensions in other packages than Dates, so it might be beneficial to keep LinearAlgebra as a weak dependency and only make Dates a regular dependency.